libs/elasticsearch/langchain_elasticsearch/_async/cache.py

```diff
@@ -0,0 +1,427 @@
+    """An Elasticsearch cache integration for LLMs.
+
+    For synchronous applications, use the ``ElasticsearchCache`` class.
+    For asyhchronous applications, use the ``AsyncElasticsearchCache`` class.
+    """
+
+    def __init__(
+        self,
+        index_name: str,
+        store_input: bool = True,
+        store_input_params: bool = True,
+        metadata: Optional[Dict[str, Any]] = None,
+        *,
+        es_url: Optional[str] = None,
+        es_cloud_id: Optional[str] = None,
+        es_user: Optional[str] = None,
+        es_api_key: Optional[str] = None,
+        es_password: Optional[str] = None,
+        es_params: Optional[Dict[str, Any]] = None,
+    ):
+        """
+        Initialize the Elasticsearch cache store by specifying the index/alias
+        to use and determining which additional information (like input, input
+        parameters, and any other metadata) should be stored in the cache.
+
+        Args:
+            index_name (str): The name of the index or the alias to use for the cache.
+                If they do not exist an index is created,
+                according to the default mapping defined by the `mapping` property.
+            store_input (bool): Whether to store the LLM input in the cache, i.e.,
+                the input prompt. Default to True.
+            store_input_params (bool): Whether to store the input parameters in the
+                cache, i.e., the LLM parameters used to generate the LLM response.
+                Default to True.
+            metadata (Optional[dict]): Additional metadata to store in the cache,
+                for filtering purposes. This must be JSON serializable in an
+                Elasticsearch document. Default to None.
+            es_url: URL of the Elasticsearch instance to connect to.
+            es_cloud_id: Cloud ID of the Elasticsearch instance to connect to.
+            es_user: Username to use when connecting to Elasticsearch.
+            es_password: Password to use when connecting to Elasticsearch.
+            es_api_key: API key to use when connecting to Elasticsearch.
+            es_params: Other parameters for the Elasticsearch client.
+        """
+
+        self._index_name = index_name
+        self._store_input = store_input
+        self._store_input_params = store_input_params
+        self._metadata = metadata
+        self._es_client = create_async_elasticsearch_client(
+            url=es_url,
+            cloud_id=es_cloud_id,
+            api_key=es_api_key,
+            username=es_user,
+            password=es_password,
+            params=es_params,
+        )
+        self._is_alias: Optional[bool] = None
+
+    async def is_alias(self) -> bool:
+        if self._is_alias is None:
+            self._is_alias = await _manage_cache_index(
+                self._es_client,
+                self._index_name,
+                self.mapping,
+            )
+        return self._is_alias  # type: ignore[return-value]
```

Comment:
Can this be called multiple times? If yes, it would benefit from caching

Reply:
This is caching the result of `_manage_cache_index()` in `self._is_alias`, so that the call is made only once. Do you mean some other type of caching beyond this?

Reply:
I totally missed this, sorry!